### DISCUSSION TITLE
fermi hubbard model (two ladder model)

### DISCUSSION THREAD

**Discussion by Yuanningss** [QuSpin/QuSpin#528] [General]
Hi 

I am now considering the fermi hubbard model (two ladder model) with the hamiltonian given by :
<img width="485" alt="static hamiltonian" src="https://user-images.githubusercontent.com/104153268/164995583-57a02a29-d715-44bc-bb25-ba4f5fff3cde.png">

Here I take chemical potential and the interaction to be zero and N (the system size) is taken to be 12. (6 sites on the ground band and 6 sites on the excited band). However, when I diagonalize the hamiltonian using eigsh and obtain the states in the ground band, I found that the for each state, the entries are all real elements. For example, for the second lowest energy state, the state is given by:
<img width="446" alt="state" src="https://user-images.githubusercontent.com/104153268/164995804-fadb7a50-fe0d-4889-b778-d05bf2b6c29c.png">

but, in principle the state should has complex coefficient (based on my hand calculation). Therefore, I was wondering if the quspin library is unable represent state with complex entries.

**Comment by weinbe58** [comment]
set `dtype=np.complex128`
